crates/codegen/grammar/src/dsl.rs

```diff
@@ -212,8 +212,8 @@ macro_rules! slang_grammar_definition {
 
     (@context_method None ) => {};
     (@context_method $context:ident ) => {
-        fn context(&self) -> &'static str {
-            stringify!($context)
+        fn lexical_context(&self) -> Option<&'static str> {
```

Comment:
Is there a case where the DSL actually accepts a `None` context? even the `Default` one has a user-defined name. I wonder why the change here?